include/video.h

```diff
@@ -146,19 +146,54 @@ enum class PixelFormat : uint8_t {
 
 	// 32K high colour, 5 bits per red/blue/green component;
 	// stored as packed uint16 data with highest bit unused
-	BGR555 = 15,
 	//
+	// Stored as array of uint16_t in host native endianess.
+	// Each uint16_t is layed out as follows:
+	// (msb)1X 5R 5G 5B(lsb)
+	// Example:
+	// uint16_t pixel = (red << 10) | (green << 5) | (blue << 0)
+	//
+	// SDL Equivalent: SDL_PIXELFORMAT_RGB555
+	// FFmpeg Equivalent: AV_PIX_FMT_RGB555
+	Packed16_RGB555 = 15,
+
 	// 65K high colour, 5 bits for red/blue, 6 bit for green;
 	// stored as packed uint16 data
-	BGR565 = 16,
 	//
+	// Stored as array of uint16_t in host native endianess.
+	// Each uint16_t is layed out as follows:
+	// (msb)5R 6G 5B(lsb)
+	// Example:
+	// uint16_t pixel = (red << 11) | (green << 5) | (blue << 0)
+	//
+	// SDL Equivalent: SDL_PIXELFORMAT_RGB565
+	// FFmpeg Equivalent: AV_PIX_FMT_RGB565
+	Packed16_RGB565 = 16,
+
 	// 16.7M (24-bit) true colour, 8 bits per red/blue/green component;
 	// stored as packed 24-bit data
-	BGR888 = 24,
 	//
+	// Stored as array of uint8_t in BGR memory order (endian agnostic)
+	// Example:
+	// uint8_t *pixels = image.image_data;
+	// pixels[0] = blue; pixels[1] = green; pixels[2] = red;
+	//
+	// SDL Equivalent: SDL_PIXELFORMAT_BGR24
+	// FFmpeg Equivalent: AV_PIX_FMT_BGR24
+	ByteArray_BGR24 = 24,
+
 	// 16.7M (32-bit) true colour; 8 bits per red/blue/green component;
-	// stored as packed uint32 data with lowest 8 bits unused
-	BGRX8888 = 32
+	// stored as packed uint32 data with highest 8 bits unused
+	//
+	// Stored as array of uint32_t in host native endianess.
+	// Each uint32_t is layed out as follows:
+	// (msb)8X 8R 8G 8B(lsb)
+	// Example:
+	// uint32_t pixel = (unused << 24) | (red << 16) | (green << 8) | (blue << 0)
+	//
+	// SDL Equivalent: SDL_PIXELFORMAT_XRGB8888
+	// FFmpeg Equivalent: AV_PIX_FMT_0RGB32
+	Packed32_XRGB8888 = 32
```

Comment:
This looks very good @weirddan455 , nice job getting to the bottom of this and documenting it! 🎉 My only minor suggestion is to flip the two parts of the enum names like this:

```
RGB555_Packed16
RGB565_Packed16
BGR24_ByteArray
XRGB8888_Packed32
```

For me that reads better as the primary info is the pixel format (e.g. BGR555), then the rest is extra information.

Reply:
Sure, that's fair.  I'll change that real quick.

Reply:
Changed